plugins/reports/monographReport/MonographReportPlugin.php

```diff
@@ -0,0 +1,409 @@
+     */
+    public function getDescription(): string
+    {
+        return __('plugins.reports.monographReport.description');
+    }
+
+    /**
+     * @copydoc ReportPlugin::display()
+     */
+    public function display($args, $request): void
+    {
+        $this->press = $this->getRequest()->getContext();
+        if (!$this->press) {
+            throw new Exception('The monograph report requires a context');
+        }
+
+        $output = $this->createOutputStream();
+
+        // Display the data rows.
+        foreach ($this as $row) {
+            $output->fputcsv($row);
+        }
+    }
+
+    /**
+     * Retrieves a row generator
+     */
+    public function getIterator(): Traversable
+    {
+        /** @var StageAssignmentDAO */
+        $stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+        /** @var UserGroup[] */
+        $userGroups = Repo::userGroup()->getCollector()->filterByContextIds([$this->press->getId()])->getMany()->toArray();
+        $editorUserGroupIds = array_map(
+            fn (UserGroup $userGroup) => $userGroup->getId(),
+            array_filter($userGroups, fn (UserGroup $userGroup) => in_array($userGroup->getRoleId(), [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR]))
+        );
+
+        $this->retrieveLimits();
+
+        $dataMapper = $this->getDataMapper();
+        yield array_keys($dataMapper);
+
+        /** @var Submission */
+        foreach (Repo::submission()->getCollector()->filterByContextIds([$this->press->getId()])->getMany() as $this->submission) {
+            $this->statusMap ??= $this->submission->getStatusMap();
+            $this->publication = $this->submission->getCurrentPublication();
+            $this->authors = $this->publication->getData('authors')->values()->toArray();
+            $this->decisionsByEditor = collect(Repo::decision()->getCollector()->filterBySubmissionIds([$this->submission->getId()])->getMany())
+                ->groupBy(fn (Decision $decision) => $decision->getData('editorId'))
+                ->toArray();
+            $this->editors = collect($stageAssignmentDao->getBySubmissionAndStageId($this->submission->getId())->toIterator())
+                ->filter(fn (StageAssignment $stageAssignment) => in_array($stageAssignment->getUserGroupId(), $editorUserGroupIds))
+                ->map(fn (StageAssignment $stageAssignment) => $this->getUser($stageAssignment->getUserId()))
+                ->unique(fn (User $user) => $user->getId())
+                ->values()
+                ->toArray();
+            yield array_map(fn (callable $callable) => $callable(), $dataMapper);
+        }
+    }
+
+    /**
+     * Retrieves the stage label
+     */
+    public function getStageLabel(int $stageId): string
+    {
+        return match ($stageId) {
+            WORKFLOW_STAGE_ID_SUBMISSION => __('submission.submission'),
+            WORKFLOW_STAGE_ID_INTERNAL_REVIEW => __('workflow.review.internalReview'),
```

Comment:
This stage is missing at the original plugin. Is there a reason?

Reply:
Yes, that is OK, only OMP has the internal review stage :-)